### DISCUSSION TITLE
Forecasting with a trained model

### DISCUSSION THREAD

**Discussion by Dreamlone** [winedarksea/AutoTS#94] [Q&A]
Hi! 
First I would like to say a big thank you for your library 😄

I apologize in advance if this information is out there, but I just couldn't find it.  Anyway, my question is: 
How can I give a prediction already trained model for the same time series on which the model was trained, but in a different section? 

For example, autoregressive models use the last n values of a time series to give a prediction. Then, to get predictions elsewhere in the same time series, I have to prepare other n values. This can be applied to the validation algorithm shown in the animation (look at the bottom part)
![sample_forecast](https://user-images.githubusercontent.com/52418437/135829991-42ed7622-3e72-4c67-82e1-a7f565bf9eb4.gif)

Is there any way I can implement such algorithm, or maybe you already have one in AutoTS?
Best regards

**Comment by winedarksea** [comment]
Thanks, and nice picture there!

I think what you want is the `model_forecast` function. You should be able to get what you want by updating the `df_train` input with the new, increased data, and just loop through. The `back_forecast` function of AutoTS does something similar, but on historic data. See more [here](https://github.com/winedarksea/AutoTS/blob/master/extended_tutorial.md#running-just-one-model)

Where `model` is a fitted AutoTS class...
You can find the model attributes here, which you can pass into the `model_forecast` function:
```
model.best_model
model.best_model_name
model.best_model_params
model.best_model_transformation_params
```
You can also check out model.results() table for other models and params that were tried

Some warnings: model_forecast doesn't sanitize the inputs as much as the AutoTS class does.

Let me know if that helps!

**Reply by Dreamlone** [reply]
Thank you so much for the quick response! 
I will try to do as you said  

Update: Thank you so much, that helped
